NDXINTER/reduce.py

```diff
@@ -1,41 +1,171 @@
-    for dir in required_dirs:
-        if not os.path.isdir(dir):
-            raise RuntimeError("Unable to find directory: {}".format(dir))
-    print("Validation successful")
-        
+    run_str = input_file.split("INTER")[1].split(".")[0].strip("0")
+    instrument='INTER'
+    name=instrument+run_str
+    ws=Load(Filename=name, OutputWorkspace='TOF_'+run_str)
+    # Filter the logs for all angles starting from time 0 and use the average of the returned angles
+    (angle_list, average_angle) = FilterLogByTime(ws, 'Theta', StartTime=0)
+    return average_angle, name
 
-def main(input_file, output_dir):
-    validate(input_file, output_dir)
-    
-    # Example of printing some stuff which is captured in autoreduction
-    # output log file 
-    print(web_var)
-    print("input_file = " + str(input_file))
-    print("output_dir = " + str(output_dir))
-    
-    # Copy raw data to output dir.
-    # Note this should only be done if raw files are small and for specific
-    # purpose such as testing 
-    shutil.copy(input_file, output_dir)
-
-    # And of course, here and below insert your reduction code!
+def parse_json_settings(angle):
+    """
+    Get experiment settings and instrument settings from JSON file
+    :param angle: Angle passed in and used to select "per angle defaults"
+    :return: Returns all of the parameters needed to do the reduction
+    """
+    json_input = r"C:\Users\wyf59278\Documents\repos\reflectometry_reduction\settings.json"
+
+    with open(json_input, "r") as read_file:
+        data = json.load(read_file)
+
+    #========================================================================================
+    # Experiment Settings
+    #========================================================================================
+
+    experimentView = data["experimentView"]
+
+    # Set a string based on what integer value is found
+    if experimentView["analysisModeComboBox"] == 1:
+        analysis_mode = "MultiDetectorAnalysis"
+    elif experimentView["analysisModeComboBox"] == 0:
+        analysis_mode = "PointDetectorAnalysis"
+    else:
+        raise Exception # If the value isn't 1 or 0 then it isn't valid
+
+    perAngleDefaults = experimentView["perAngleDefaults"]
+    rows = perAngleDefaults["rows"]
+
+    # This looks for the run angle and set other parameters accordingly
+    # Using a tolerance of +-0.5% of the motor readback angle
+    min = angle * 0.995
+    max = angle * 1.005
+    angle_found = False
+    for i in rows:
+        # If the value is within -0.5% to +0.5% it is counted as a match
+        if min <= float(i[0]) <= max:
+            angle_found = True
+            first_transmission_run_list = i[1]
+            second_transmission_run_list = i[2]
+            transmission_processing_instructions = i[3]
+            # Skipping over parameters that are present in the JSON file but not currently used in the reduction
+            processing_instructions = i[8]

```

Comment:
You can move this into a little helper function, e.g. `get_transmission_params` that takes `i` and sets the correct fields. This would make it easier to test as well, because now a unit test can be added for the helper function to make sure it works